src/lib/dns.coffee

```diff
@@ -114,42 +112,27 @@ module.exports = (dnschain) ->
             # At some point we may still want to support this though.
             q = req.question[qIdx=0]
             q.name = q.name.toLowerCase()
-            
+
             ttl = Math.floor(Math.random() * 3600) + 30 # TODO: pick an appropriate TTL value!
             @log.debug "received question", q
 
-            if /\.(bit|p2p)$/.test q.name
-                if S(q.name).endsWith '.bit'
-                    nmcDomain = @namecoinizeDomain q.name
-                    resolver = 'nmc'
-                else
-                    # TODO: bdns-izeDomain
-                    nmcDomain = S(q.name).chompRight('.p2p').s
-                    resolver = 'bdns'
-                
-                @log.debug gLineInfo("resolving via #{resolver}..."), {nmcDomain:nmcDomain, q:q}
-
-                @dnschain[resolver].resolve nmcDomain, (err, result) =>
-                    # @log.warn gLineInfo('blah!'), {result: result}
+            chainTLDs = new RegExp('\\.('+_.map(@dnschain.chains, 'tld').join('|')+')$')
+            if chainTLDs.test q.name
+                tld = q.name.split('.').pop()
+                resolver = _.find @dnschain.chains, (c) ->
+                   c.tld? and c.tld == tld
+
+                @log.debug gLineInfo("resolving via #{resolver.name}..."), {domain:q.name, q:q}
+
+                resolver.resolve q.name, {}, (err, result) =>
                     if err? or !result
-                        @log.error gLineInfo("#{resolver} failed to resolve"), {err:err?.message, result:result, q:q}
+                        @log.error gLineInfo("#{resolver.name} failed to resolve"), {err:err?.message, result:result, q:q}
                         @sendErr res, null, cb
                     else
-                        @log.debug gLineInfo("#{resolver} resolved query"), {q:q, d:nmcDomain, result:result}
-
-                        try
-                            # result.value = JSON.parse result.value
-                            # TODO: [BDNS] this! also, note that we're converting JSON multiple times
-                            #       this is ineffecient and ugly.
-                            #       organize all this code in a generic way to support all blockchains.
-                            result.value = @dnschain[resolver].toJSONobj result
-                        catch e
-                            @log.warn e.stack
-                            @log.warn gLineInfo("bad JSON!"), {q:q, result:result}
-                            return @sendErr res, NAME_RCODE.FORMERR, cb
-
-                        if not (handler = dnsTypeHandlers.namecoin[QTYPE_NAME[q.type]])
-                            @log.warn gLineInfo("no such handler!"), {q:q, type: QTYPE_NAME[q.type]}
+                        @log.debug gLineInfo("#{resolver.name} resolved query"), {q:q, d:q.name, result:result}
+
+                        if not (handler = @dnschain.chains[resolver.name].dnsHandler[QTYPE_NAME[q.type]])
```

Comment:
Why are you doing `@dnschain.chains[resolver.name]`? Can't you just do `resolver.dnsHandler`?